lib/common/Memory/SmallFinalizableHeapBlock.h

```diff
@@ -144,25 +145,29 @@ class SmallFinalizableWithBarrierHeapBlockT : public SmallFinalizableHeapBlockT<
 
     virtual bool FindHeapObject(void* objectAddress, Recycler * recycler, FindHeapObjectFlags flags, RecyclerHeapObjectInfo& heapObject) override sealed
     {
-        return FindHeapObjectImpl<SmallFinalizableWithBarrierHeapBlockT>(objectAddress, recycler, flags, heapObject);
+        return this->template FindHeapObjectImpl<SmallFinalizableWithBarrierHeapBlockT<TBlockAttributes>>(objectAddress, recycler, flags, heapObject);
     }
 protected:
     SmallFinalizableWithBarrierHeapBlockT(HeapBucketT<SmallFinalizableWithBarrierHeapBlockT> * bucket, ushort objectSize, ushort objectCount)
-        : SmallFinalizableHeapBlockT(bucket, objectSize, objectCount, TBlockAttributes::IsSmallBlock ? SmallFinalizableBlockWithBarrierType : MediumFinalizableBlockWithBarrierType)
+        : SmallFinalizableHeapBlockT<TBlockAttributes>(bucket, objectSize, objectCount, TBlockAttributes::IsSmallBlock ? Base::SmallFinalizableBlockWithBarrierType : Base::MediumFinalizableBlockWithBarrierType)
     {
     }
 };
 #endif
 
+#if defined(_MSC_VER) || !defined(SMALL_FINALIZABLE_HEAP_BLOCK_CPP)
 extern template class SmallFinalizableHeapBlockT<SmallAllocationBlockAttributes>;
 extern template class SmallFinalizableHeapBlockT<MediumAllocationBlockAttributes>;
```

Comment:
I tried deleting these ```extern template class ...``` lines, both VS and clang builds without problem. If this works, can revert ```+#define SMALL_FINALIZABLE_HEAP_BLOCK_CPP``` and vcproj ```NotUsing``` precompiled header  workarounds.